core/sync/sync_windows.odin

```diff
@@ -1,51 +1,41 @@
+// +build windows
 package sync
 
 import "core:sys/win32"
 
-foreign {
-	@(link_name="llvm.x86.sse2.pause")
-	yield_processor :: proc() ---
-}
-
-Semaphore :: struct {
-	_handle: win32.Handle,
-}
-
+// A lock that can only be held by one thread at once.
 Mutex :: struct {
 	_critical_section: win32.Critical_Section,
 }
 
+// Blocks until signalled.
+// When signalled, awakens exactly one waiting thread.
 Condition :: struct {
 	event: win32.Handle,
 }
 
-Ticket_Mutex :: struct {
-	ticket:  u64,
-	serving: u64,
+// When waited upon, blocks until the internal count is greater than zero, then subtracts one.
+// Posting to the semaphore increases the count by one, or the provided amount.
+Semaphore :: struct {
+	_handle: win32.Handle,
 }
 
 
-current_thread_id :: proc() -> i32 {
-	return i32(win32.get_current_thread_id());
-}
-
-semaphore_init :: proc(s: ^Semaphore) {
-	s._handle = win32.create_semaphore_w(nil, 0, 1<<31-1, nil);
+semaphore_init :: proc(s: ^Semaphore, initial_count := 0) {
+	assert(initial_count >= min(int) && initial_count <= max(int));
```

Comment:
Could you say
```odin
semaphore_init :: proc(s: ^Semaphore, initial_count : int = 0) { ...
```
instead of this assertion? If so, would this be desirable?
This could possibly be extended to the other implementations as well if desirable.

Reply:
That assertion makes no sense either as it will always be true.

Reply:
Quite right. I meant to assert that the count was within range of `i32` since I'm about to cast to that, and that cast doesn't check.

Reply:
This now reads
```
assert(initial_count >= min(i32) && initial_count <= max(i32));
```

Reply:
I'm open to better ways to do it though.

Reply:
Okay yeah - I don't know how to best handle this - I want to say something like:
```
assert(is_within_range(initial_count, i32));
```

Reply:
Although I guess we don't really want an assert on a common path; `semaphore_post` has a similar problem with `count` being an `i32`.
Sigh.